### PR TITLE
Add argument to disable yaml cronjob when factsource => yaml

### PR DESCRIPTION
**Pull request by kronos-pbrideau**
The `$::path` variable was messing a lot with multiple distributions...

```
puppet agent -t  on centos : PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/root/bin
puppet agent -t on debian: PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
automated run (/etc/init.d/puppet) centos: PATH=/usr/bin:/sbin:/bin:/usr/sbin
automated run (/etc/init.d/puppet) debian: PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
```
(packages from puppetlabs: 3.6.2-1.el6 and 3.6.2-1puppetlabs1)

### CODE REVIEW

**Comment by kwolf** [comment]
You guys planning on taking this one? It's the only thing that's keeping me from using the puppetlabs repo.

**Comment by ianau74** [comment]
is there a way we can move forward with this?  The current refresh-mcollective-metadata cronjob is broken.  Here's a JIRA: https://tickets.puppetlabs.com/browse/MODULES-1699

**Comment by ffrank** [comment]
Thanks for this patch! I'm reviewing it now and will leave some notes for discussion.

File: spec/classes/mcollective_spec.rb [hunk 1]
```diff
@@ -145,6 +145,17 @@
...
           end
         end
+
+        describe '#yaml_fact_cron' do
+          context 'default (true)' do
+            it { should contain_cron('refresh-mcollective-metadata') }
+          end
+
+          context 'false' do
+            let(:params) { { :server => false } }
```
Comment (ffrank) [inline spec/classes/mcollective_spec.rb]: Why is this `:server` as opposed to `:yaml_fact_cron`?

I suspect that this is a simple copy/paste error, but then I really wonder how these ever passed.

**Comment by kronos-pbrideau** [comment]
Yep, it was definitely a typo, I pushed --force to omit a simple commit named typo

**Comment by nibalizer** [comment]
Looks like this has merge conflicts but other than that LGTM

**Comment by kronos-pbrideau** [comment]
Sorry, i'm not very used to this, and still learning the git workflow with upstream branches and so on.  I'll be cleaner next time!

**Comment by ffrank** [comment]
No problem, I'll clean this up for you this time. (You pretty much never want to have merge commits in topic branches, `rebase` is absolutely preferable.)

I will also look into why the tests didn't fail before the fix.

**Comment by ffrank** [comment]
Closing this in favor of #216.

**Pull request closed without merging**